### PR TITLE
feat: Add Support For Mixed Photo + Video Library Selection [CC-495]

### PR DESCRIPTION
**Pull request by zeph-cohen10**
This PR works off of https://github.com/rewardStyle/YPImagePicker/pull/26 as a starting point which adds new functionality to select both videos and images from the library. This PR adds a new property `allowPhotoAndVideoSelection` on `YPConfigLibrary` which can be toggled true/false from the client. When `allowPhotoAndVideoSelection` is set to true the client will be able to make mixed selection from the library.

### CODE REVIEW

File: Source/Pages/Gallery/YPLibraryVC+CollectionView.swift [hunk 1]
```diff
@@ -144,9 +144,18 @@ extension YPLibraryVC: UICollectionViewDelegate {
...
+            cell.isUserInteractionEnabled = !(isMultipleSelectionEnabled && isVideo)
+        } else {
+            cell.multipleSelectionIndicator.isHidden = !isMultipleSelectionEnabled
+            cell.isUserInteractionEnabled = true
+        }
+
+
         cell.isSelected = !disableAutomaticCellSelection && currentlySelectedIndex == indexPath.row && selectedItems.contains(where: { $0.assetIdentifier == asset.localIdentifier })
-        cell.isUserInteractionEnabled = !(isMultipleSelectionEnabled && isVideo)
+        cell.isUserInteractionEnabled = true
```
Comment (MikkoKuivanenRS) [inline Source/Pages/Gallery/YPLibraryVC+CollectionView.swift:158]: Should this line be removed since it is already set above

File: Source/Pages/Gallery/LibraryMediaManager.swift [hunk 2]
```diff
@@ -225,6 +233,7 @@ public class LibraryMediaManager {
                     videoCompositionTrack.preferredTransform = transform
                 }
 
+
```
Comment (will-ltk) [inline Source/Pages/Gallery/LibraryMediaManager.swift:236]: Need this?

File: Source/Pages/Gallery/YPLibraryVC+CollectionView.swift (hunk 1 above)
Comment (will-ltk) [reply Source/Pages/Gallery/YPLibraryVC+CollectionView.swift:158]: +1

**Review by MikkoKuivanenRS — APPROVED** [review]

File: Source/Pages/Gallery/LibraryMediaManager.swift [hunk 3]
```diff
@@ -197,7 +199,13 @@ public class LibraryMediaManager {
 
                 // 5. Configuring export session
                 let videoIsCropped = cropRect.size.width < abs(videoSize.width) || cropRect.size.height < abs(videoSize.height)
-                let presetName = compressionTypeOverride ?? (videoIsCropped || videoIsTrimmed || videoIsRotated || (shouldMute && videoNeedsProcessing) ? YPConfig.video.compression : AVAssetExportPresetPassthrough)
+
+                var presetName: String = ""
+                if !YPImagePickerConfiguration.shared.library.allowPhotoAndVideoSelection {
+                    presetName = compressionTypeOverride ?? (videoIsCropped || videoIsTrimmed || videoIsRotated || (shouldMute && videoNeedsProcessing) ? YPConfig.video.compression : AVAssetExportPresetPassthrough)
+                } else {
+                    presetName = AVAssetExportPresetHighestQuality
```
Comment (MikkoKuivanenRS) [inline Source/Pages/Gallery/LibraryMediaManager.swift:207]: Where did `AVAssetExportPresetHighestQuality` come from? Seems unrelated to what was here before.

File: Source/Pages/Gallery/LibraryMediaManager.swift [hunk 4]
```diff
@@ -183,10 +183,12 @@ public class LibraryMediaManager {
...
-                do {
-                    try videoCompositionTrack.insertTimeRange(trackTimeRange, of: videoTrack, at: CMTime.zero)
-                } catch {
-                    ypLog("Unexpected error: \(error).")
+                if !YPImagePickerConfiguration.shared.library.allowPhotoAndVideoSelection {
+                    do {
+                        try videoCompositionTrack.insertTimeRange(trackTimeRange, of: videoTrack, at: CMTime.zero)
+                    } catch {
+                        ypLog("Unexpected error: \(error).")
+                    }
```
Comment (MikkoKuivanenRS) [inline Source/Pages/Gallery/LibraryMediaManager.swift:191]: What was the reason behind this?

Comment (zeph-cohen10) [reply Source/Pages/Gallery/LibraryMediaManager.swift]: This change doesn't seem necessary, I think it came from the WIP branch but it's not strictly necessary for multi video selection when I tested without this change. I removed it here: https://github.com/rewardStyle/YPImagePicker/pull/27/commits/d11a88a5c675d3787e42e42e08037e91bba1cb99 @MikkoKuivanenRS

File: Source/Pages/Gallery/LibraryMediaManager.swift (hunk 3 above)
Comment (zeph-cohen10) [reply Source/Pages/Gallery/LibraryMediaManager.swift]: Yeah it was on the WIP branch but tested without this change and I didn't see any issues with the multi video selection. Removed it here: https://github.com/rewardStyle/YPImagePicker/pull/27/commits/d11a88a5c675d3787e42e42e08037e91bba1cb99

**Review by MikkoKuivanenRS — APPROVED** [review]

**Review by will-ltk — APPROVED** [review]
